Classes/Emogrifier.php

```diff
@@ -537,20 +548,36 @@ private function generateStyleStringFromDeclarationsArrays(array $oldStyles, arr
         return trim($style);
     }
 
-
     /**
      * Copies the media part from CSS array parts to $xmlDocument.
      *
-     * @param string[] $cssParts
-     * @param \DOMDocument $xmlDocument
+     * @param string $mediaCss A string of CSS.
+     * @param \DOMDocument $xmlDocument The document to match against.
      *
      * @return void
      */
-    public function copyCssWithMediaToStyleNode(array $cssParts, \DOMDocument $xmlDocument)
+    public function copyCssWithMediaToStyleNode($mediaCss, \DOMDocument $xmlDocument)
     {
-        if (isset($cssParts['media']) && $cssParts['media'] !== '') {
-            $this->addStyleElementToDocument($xmlDocument, $cssParts['media']);
+        if (empty($mediaCss)) {
```

Comment:
This is a string. So the check should be !== '', not empty (as empty on strings will provide false positives).